### PR TITLE
docs: Include Vercel CLI Deployment Instructions

### PR DESCRIPTION
**Pull request by ajcwebdev**
## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

- Added instructions for deploying to Vercel from the CLI.
- Added information about configuring your project with `vercel.json`.
- Remove superfluous information in the original Astro docs template repo.
- Include docs for contributors wanting to run the docs site locally. 

---

## Screenshots

Check the deploy preview.

💯

### CODE REVIEW

File: www/src/pages/en/deployment/vercel.md [hunk 1]
```diff
@@ -6,19 +6,47 @@ layout: ../../../layouts/docs.astro
...
 
-4. Import the GitHub repository with your project.
+```json
+{
+  "buildCommand": "pnpm build",
+  "outputDirectory": "dist",
+  "devCommand": "pnpm dev",
+  "installCommand": "pnpm install"
+}
+```
```
Comment (nexxeln) [inline www/src/pages/en/deployment/vercel.md:20]: Should use `npm` in the docs as its the standard.

File: www/src/pages/en/deployment/vercel.md [hunk 2]
```diff
@@ -6,19 +6,47 @@ layout: ../../../layouts/docs.astro
...
+
+4. Click **Deploy**. Now whenever you push a change to your repository, Vercel will automatically redeploy your website!
+
+## Using the Vercel CLI
+
+To deploy from the command line you can either:
+
+- [Install the Vercel CLI globally](https://vercel.com/docs/cli#installing-vercel-cli) with `pnpm i -g vercel`.
+
+- Add the CLI as a dependency to your project with `pnpm add -D vercel` and run commands with `pnpm vercel`.
```
Comment (nexxeln) [inline www/src/pages/en/deployment/vercel.md:44]: Is this step necessary? Can you not just use the global cli?

File: www/src/pages/en/deployment/vercel.md [hunk 3]
```diff
@@ -6,19 +6,47 @@ layout: ../../../layouts/docs.astro
...
+
+Include `--env DATABASE_URL=YOUR_DATABASE_URL_HERE` for environment variables like the database connection string. Use `--confirm` if you want to skip the deployment questions and give the default answer for each.
+
+```bash
+vercel
+```
+
+After the first deployment this command will deploy to a preview branch. You will need to include `--prod` to push changes directly to the live site for future deployments.
```
Comment (nexxeln) [inline www/src/pages/en/deployment/vercel.md]: After the first deployment this command will deploy to a preview branch. To push changes directly to the live site include `--prod`.

```bash
vercel --prod
```

File: www/src/pages/en/deployment/vercel.md [hunk 4]
```diff
@@ -6,19 +6,47 @@ layout: ../../../layouts/docs.astro
...
+
+## Using the Vercel CLI
+
+To deploy from the command line you can either:
+
+- [Install the Vercel CLI globally](https://vercel.com/docs/cli#installing-vercel-cli) with `pnpm i -g vercel`.
+
+- Add the CLI as a dependency to your project with `pnpm add -D vercel` and run commands with `pnpm vercel`.
+
+Include `--env DATABASE_URL=YOUR_DATABASE_URL_HERE` for environment variables like the database connection string. Use `--confirm` if you want to skip the deployment questions and give the default answer for each.
```
Comment (nexxeln) [inline www/src/pages/en/deployment/vercel.md:46]: Show an example command too.

**Review by nexxeln — CHANGES_REQUESTED** [review]

File: www/src/pages/en/deployment/vercel.md (hunk 2 above)
Comment (ajcwebdev) [reply www/src/pages/en/deployment/vercel.md:44]: It's a philosophical debate. Some people prefer avoiding global installs cause then everyone gets the same version and also you can just clone down the repo and run the command without needing to worry about installing anything onto your machine which can be a pain when there's a bunch of different ways to install a CLI. In fact Vercel's command for installing globally with `pnpm` is actually incorrect right now and says `pnpm i -g vercel`.

**Review by nexxeln — APPROVED** [review]
Amazing

**Pull request merged**